src/node/node_state.h

```diff
@@ -561,6 +563,8 @@ namespace ccf
               n2n_channels_cert);
             setup_progress_tracker();
             setup_history();
+            if (endorsed_node_cert.has_value())
```

Comment:
This is only for a 2.x node joining a 1.x service. It'd be nicer if it was in the `if` branch on line 539 the history isn't setup yet but a comment here would be useful?

Reply:
Yes, cleaned up.